service/cachemdw/keys.go

```diff
@@ -0,0 +1,67 @@
+package cachemdw
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/kava-labs/kava-proxy-service/decode"
+)
+
+type CacheItemType int
+
+const (
+	CacheItemTypeQuery CacheItemType = iota + 1
+)
+
+func (t CacheItemType) String() string {
+	switch t {
+	case CacheItemTypeQuery:
+		return "query"
+	default:
+		return "unknown"
+	}
+}
+
+func BuildCacheKey(cacheItemType CacheItemType, parts []string) string {
+	fullParts := append(
+		[]string{
+			cacheItemType.String(),
+		},
+		parts...,
+	)
+
+	return strings.Join(fullParts, ":")
+}
+
+// GetQueryKey calculates cache key for request
+func GetQueryKey(
+	chainID string,
```

Comment:
another vote for renaming from chainID to something like `CacheKeyPrefix`

Reply:
done